### PR TITLE
Fix tests on import modal

### PR DESCRIPTION
**Pull request by Jordanlelay**
## Work performed

- Fix tests on import button
- Fix tests on import images modal

### CODE REVIEW

File: typescript/web/src/components/import-button/import-button.fixtures.ts [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+export const ERROR_MOCKS: ApolloMockResponses = [
+  GET_DATASET_BY_SLUG_MOCK,
+  GET_WORKSPACE_ID_MOCK,
+  GET_UPLOAD_TARGET_MOCK,
+  DATASET_IMAGES_PAGE_DATASET_MOCK,
+  WORKSPACE_DATASETS_PAGE_DATASETS_MOCK,
+  ERROR_CREATE_IMAGES_MOCK,
+];
+export const IMPORT_BUTTON_MOCKS: ApolloMockResponses = [
```
Comment (bbenoist) [inline typescript/web/src/components/import-button/import-button.fixtures.ts:184]:
```suggestion
];

export const IMPORT_BUTTON_MOCKS: ApolloMockResponses = [
```

File: typescript/web/src/components/import-button/import-button.fixtures.ts [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+          images: [],
+          imagesAggregates: {
+            totalCount: 2,
+          },
+          labelClassesAggregates: {
+            totalCount: 0,
+          },
+          labelsAggregates: {
+            totalCount: 0,
+          },
```
Comment (bbenoist) [inline typescript/web/src/components/import-button/import-button.fixtures.ts:140]:
```suggestion
          imagesAggregates: { totalCount: 2 },
          labelClassesAggregates: { totalCount: 0 },
          labelsAggregates: { totalCount: 0 },
```

File: typescript/web/src/components/import-button/import-button.fixtures.ts [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+      workspaceSlug: BASIC_DATASET_DATA.workspace.slug,
+    },
+  },
+  result: {
+    data: {
+      dataset: {
+        ...BASIC_DATASET_DATA,
+        imagesAggregates: {
+          totalCount: 2,
+        },
```
Comment (bbenoist) [inline typescript/web/src/components/import-button/import-button.fixtures.ts:112]:
```suggestion
        imagesAggregates: { totalCount: 2 },
```

File: typescript/web/src/components/import-button/import-images-modal/__tests__/import-images-modal.tsx [hunk 4]
```diff
@@ -1,96 +1,63 @@
...
 
-function renderModalAndImport(filesToImport = files, props = {}) {
+const renderModalAndImport = (
+  filesToImport = files,
+  props = {},
+  error = false
+) => {
   render(<ImportImagesModal isOpen onClose={() => {}} {...props} />, {
-    wrapper: Wrapper,
+    wrapper: ({ children }) => Wrapper({ error, children }),
```
Comment (bbenoist) [inline typescript/web/src/components/import-button/import-images-modal/__tests__/import-images-modal.tsx:55]:
```suggestion
    wrapper: createWrapper(error),
```

File: typescript/web/src/components/import-button/import-images-modal/__tests__/import-images-modal.tsx [hunk 5]
```diff
@@ -1,96 +1,63 @@
...
+type WrapperProps = PropsWithChildren<{
+  error?: boolean;
+}>;
+const Wrapper = ({ error = false, children }: WrapperProps) => (
+  <MockedProvider
+    link={getApolloMockLink(error ? ERROR_MOCKS : IMPORT_BUTTON_MOCKS)}
+  >
+    {children}
+  </MockedProvider>
 );
```
Comment (bbenoist) [inline typescript/web/src/components/import-button/import-images-modal/__tests__/import-images-modal.tsx:41]:
```suggestion
const createWrapper =
  (error: boolean = false) =>
  ({ children }: PropsWithChildren<{}>) => (
    <MockedProvider
      link={getApolloMockLink(error ? ERROR_MOCKS : IMPORT_BUTTON_MOCKS)}
    >
      {children}
    </MockedProvider>
  );
```

**Review by bbenoist — CHANGES_REQUESTED** [review]

File: typescript/web/src/components/import-button/import-images-modal/__tests__/import-images-modal.tsx (hunk 5 above)
Comment (Jordanlelay) [reply typescript/web/src/components/import-button/import-images-modal/__tests__/import-images-modal.tsx:41]: Why would it be better to use this form ?

Comment (bbenoist) [reply typescript/web/src/components/import-button/import-images-modal/__tests__/import-images-modal.tsx:41]: It keeps the wrapper being a React component with only a single `children` prop and prevents calling it like a function as you did after (hence the other code suggestion).

**Review by bbenoist — APPROVED** [review]

**Pull request merged**